2_process/src/process_snow_rasters.R

```diff
@@ -1,41 +1,73 @@
 
 process_snow_raster <- function(ind_file, snow_data_ind, snow_data_yml, crop_extent_raster_ind, proj_str) {
-  
+
   snow_data_fn <- sc_retrieve(snow_data_ind, snow_data_yml)
   crop_extent_sfcpolygon <- readRDS(sc_retrieve(crop_extent_raster_ind))
-  
+
   # Size of SNODAS data
   n_col <- 8192
-  n_row <- 4096 
-  
+  n_row <- 4096
+
   # boundary extent for unmasked SNODAS
   x0 <- -130.516666666661
   x1 <- -62.2499999999975
   y0 <- 24.0999999999990
   y1 <- 58.2333333333310
-  
-  snow_depth <- readBin(snow_data_fn, integer(), 
+
+  snow_depth <- readBin(snow_data_fn, integer(),
                         n=n_row*n_col, size=2, signed=TRUE, endian='big')
   snow_depth[snow_depth <= 0] <- as.numeric(NA)
   snow_depth_mat <- matrix(snow_depth, nrow = n_col)
-  
+
   # Convert to raster
   snow_raster <- raster::raster(ncol=n_col, nrow=n_row, xmn = x0, xmx = x1, ymn = y0, ymx = y1)
   snow_raster <- raster::setValues(snow_raster, snow_depth)
-  
+
   # Add the correct projection
   # Source for WGS 84 projection: "Projecting SNODAS Data" section on https://nsidc.org/data/g02158
   raster::crs(snow_raster) <- "+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs"
-  
+
   # Now reproject for this visualization
   snow_raster_proj <- raster::projectRaster(snow_raster, crs = raster::crs(proj_str))
-  
+
   # Now crop
   crop_extent_geom <- sf::st_sf(sf::st_geometry(crop_extent_sfcpolygon))
   snow_raster_proj_crop <- raster::crop(snow_raster_proj, crop_extent_geom)
-  
+
   data_file <- as_data_file(ind_file)
   saveRDS(snow_raster_proj_crop, data_file)
   gd_put(remote_ind=ind_file, local_source=data_file, mock_get='none')
-  
+
+}
+
+interpolate_snow_raster_layers <- function(timestep_in_hours, ...) {
+
+  snow_raster_rds_inds <- c(...) #would be either passing in dates or files here?
```

Comment:
I think you could pass in `'1_fetch/log/1_fetch_snow_tasks.ind'` which looks like this:

```
2_process/out/raster_snow_20190312.rds.ind: ccbaa2f343760d20d1a846a5b57049a2
2_process/out/raster_snow_20190313.rds.ind: 3f8ca1de5d72d6ce7ef525813c3c21cd
2_process/out/raster_snow_20190314.rds.ind: ce9e9b7eb900a7668d7aa96286c5e2a8
2_process/out/raster_snow_20190315.rds.ind: a5e779e55d1d1c3a4d4d2d589a8f552d
2_process/out/raster_snow_20190316.rds.ind: 822d8c7e2fa0ce8753f7fdbd45c2ac58
2_process/out/raster_snow_20190317.rds.ind: 0ef163be6308485bbbf9e77da02aa52b
2_process/out/raster_snow_20190318.rds.ind: d6a56c66f4d819c0642e0927aef09dc8
2_process/out/raster_snow_20190319.rds.ind: a315d56862bb6bee432275567abfd587
```

and you can get all the filenames by reading it in as yaml:

```
snow_raster_rds_inds <- names(yaml::yaml.load_file('1_fetch/log/1_fetch_snow_tasks.ind'))
snow_raster_rds_inds

[1] "2_process/out/raster_snow_20190312.rds.ind" "2_process/out/raster_snow_20190313.rds.ind"
[3] "2_process/out/raster_snow_20190314.rds.ind" "2_process/out/raster_snow_20190315.rds.ind"
[5] "2_process/out/raster_snow_20190316.rds.ind" "2_process/out/raster_snow_20190317.rds.ind"
[7] "2_process/out/raster_snow_20190318.rds.ind" "2_process/out/raster_snow_20190319.rds.ind"
```